File: leetcode_js/src/containsDuplicateIII.js
const containsNearbyAlmostDuplicate = function (nums, k, t) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = 1; j <= k; j++) {
      if (Math.abs(nums[i] - nums[i + j]) <= t) {
        return true;
      }
    }
  }

  return false;
};

module.exports = { containsNearbyAlmostDuplicate };

File: leetcode_go/problems/0220_contains_duplicate_iii/contains_duplicate_iii.go
package problem0220

import "math"

func containsNearbyAlmostDuplicate(nums []int, k int, t int) bool {
	for i := 0; i < len(nums); i++ {
		for j := 1; j <= k; j++ {
			if i+j < len(nums) &&
				int(math.Abs(float64(nums[i]-nums[i+j]))) <= t {
				return true
			}
		}
	}

	return false
}
